src/Features/CSharp/Portable/SignatureHelp/CSharpSignatureHelpService.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.SignatureHelp;
+
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp;
+
+[ExportLanguageServiceFactory(typeof(SignatureHelpService), LanguageNames.CSharp), Shared]
+internal class CSharpSignatureHelpServiceFactory : ILanguageServiceFactory
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpSignatureHelpServiceFactory()
+    {
+    }
+
+    public ILanguageService CreateLanguageService(HostLanguageServices languageServices)
+        => new CSharpSignatureHelpService(languageServices.LanguageServices);
+}
+
+internal class CSharpSignatureHelpService : SignatureHelpServiceWithProviders
```

Comment:
make nested type (and private).

Reply:
this type doesn't seem to serve a purpose.  make base type non-abstract and selaed.  and have the factory instantiate that instead.